tasks/test_results_processor.py

```diff
@@ -107,167 +126,147 @@ def _bulk_write_tests_to_db(
-                        else 0,
-                        "skip_count": 1 if outcome == str(Outcome.Skip) else 0,
-                        "branch": branch,
-                        "date": date.today(),
-                        "latest_run": datetime.now(),
-                        "commits_where_fail": [commitid]
-                        if (
-                            outcome == str(Outcome.Failure)
-                            or outcome == str(Outcome.Error)
-                        )
-                        else [],
-                    }
-
-                if outcome != str(Outcome.Skip):
-                    if test_id in daily_totals:
-                        update_daily_total()
-                    else:
-                        create_daily_total()
-
-            # Save Tests
-            insert_on_conflict_do_nothing = (
-                insert(Test.__table__).values(test_data).on_conflict_do_nothing()
-            )
-            db_session.execute(insert_on_conflict_do_nothing)
-            db_session.flush()
-
-            # Upsert Daily Test Totals
-            stmt = insert(DailyTestRollup.__table__).values(list(daily_totals.values()))
-            stmt = stmt.on_conflict_do_update(
-                index_elements=[
-                    "repoid",
-                    "branch",
-                    "test_id",
-                    "date",
-                ],
-                set_={
-                    "last_duration_seconds": stmt.excluded.last_duration_seconds,
-                    "avg_duration_seconds": (
-                        DailyTestRollup.__table__.c.avg_duration_seconds
-                        * (
-                            DailyTestRollup.__table__.c.pass_count
-                            + DailyTestRollup.__table__.c.fail_count
-                        )
-                        + stmt.excluded.avg_duration_seconds
+                if outcome == str(Outcome.Pass):
+                    daily_totals[test_id]["pass_count"] += 1
+                elif outcome == str(Outcome.Failure) or outcome == str(Outcome.Error):
+                    daily_totals[test_id]["fail_count"] += 1
+                elif outcome == str(Outcome.Skip):
+                    daily_totals[test_id]["skip_count"] += 1
+
+            def create_daily_total():
+                daily_totals[test_id] = {
+                    "test_id": test_id,
+                    "repoid": repoid,
+                    "last_duration_seconds": duration_seconds,
+                    "avg_duration_seconds": duration_seconds,
+                    "pass_count": 1 if outcome == str(Outcome.Pass) else 0,
+                    "fail_count": 1
+                    if outcome == str(Outcome.Failure) or outcome == str(Outcome.Error)
+                    else 0,
+                    "skip_count": 1 if outcome == str(Outcome.Skip) else 0,
+                    "flaky_fail_count": 1
+                    if test_id in flaky_test_set and outcome == str(Outcome.Failure)
```

Comment:
would an error be considered a flake too?

Reply:
yeah for flaky test detection we consider an error a failure, me omitting Error here is a mistake